File: 0525-contiguous-array/0525-contiguous-array.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        unordered_map<int,int> mp;
        int ans = 0, zero = 0, one = 0;

        for(int i = 0; i < nums.size(); i++) {

            if(nums[i] == 0) zero++;
            else one++;

            int elem = zero - one;   // ❌ removed abs()

            if(elem == 0) {
                ans = max(ans, i + 1);  // ❌ fixed length
            }
            else if(mp.find(elem) == mp.end()) {
                mp[elem] = i;          // store first index
            }
            else {
                ans = max(ans, i - mp[elem]);
            }
        }

        return ans;
    }
};

File: 0525-contiguous-array/0525-contiguous-array.java
class Solution {
    public int findMaxLength(int[] nums) {
        int[] freq = new int[2];
        Map<Integer, Integer> mp = new HashMap<>();
        int ans = 0, r=0, l=0;
        int sum = 0;
        mp.put(0,-1);
        for(int i=0;i<nums.length;i++){
           if(nums[i] == 0) nums[i] = -1;
        }
        while(r < nums.length){
            sum += nums[r];

           if(mp.containsKey(sum)){
            ans = Math.max(ans , r - mp.get(sum));
           }
           else{
              mp.put(sum, r);
           }
          
          
           r++;
        }

        return ans;
    }
}